### PR TITLE
Add readme for extending webpack config

### PR DESCRIPTION
**Pull request by animesh10**
Added how the folder should look like and a sample webpack config.

### CODE REVIEW

**Comment by karlhorky** [comment]
It looks like this `{% raw %}` directive doesn't seem to work in GitHub Flavored Markdown:

screen shot 2017-04-18 at 11 10 12

https://github.com/animesh10/electrode/blob/9ff25e26741ac972cc855da39abb5730204107ee/packages/electrode-archetype-react-app/README.md

Maybe a code block instead?

File: packages/electrode-archetype-react-app/README.md [hunk 1]
```diff
@@ -236,6 +236,35 @@ var supports = require("electrode-archetype-react-app/supports");
...
+var BellOnBundlerErrorPlugin = require('bell-on-bundler-error-plugin')
+
+const config = {
+  plugins: [
+    new BellOnBundlerErrorPlugin()
+  ]
+};
+
+module.exports = config;
+};
```
Comment (karlhorky) [inline packages/electrode-archetype-react-app/README.md]: I guess we don't need this curly brace and semicolon?

File: packages/electrode-archetype-react-app/README.md [hunk 2]
```diff
@@ -236,6 +236,35 @@ var supports = require("electrode-archetype-react-app/supports");
...
+└── config
+    └── webpack
+        └── webpack.config.js
+
+{% endraw %}
+
+Where `webpack.config.js` is your extended or overriding webpack config, for example:
+
+```js
+var BellOnBundlerErrorPlugin = require('bell-on-bundler-error-plugin')
```
Comment (karlhorky) [inline packages/electrode-archetype-react-app/README.md]: Do we want semicolons everywhere?

Comment (karlhorky) [inline packages/electrode-archetype-react-app/README.md]: Should this be a `const` to be consistent with the code style below?

File: packages/electrode-archetype-react-app/README.md [hunk 3]
```diff
@@ -236,6 +236,35 @@ var supports = require("electrode-archetype-react-app/supports");
 
 supports.cssModuleHook();
 ```
+## Extending Webpack Configuration
+
+The webpack config for your app is being generated by the archetype. While composing the webpack config, the archetype looks for the `archetype/config` folder at the project root of your app.
+In order to override or extend the webpack configuration you should create a file under `archetype/config/webpack`.
+Your directory structure should look like this:
```
Comment (karlhorky) [inline packages/electrode-archetype-react-app/README.md]: Extra empty lines between these three lines would make them into separate paragraphs, which would help with readability:

```diff
The webpack config for your app is being generated by the archetype. While composing the webpack config, the archetype looks for the `archetype/config` folder at the project root of your app.
+                                                                                                    
In order to override or extend the webpack configuration you should create a file under `archetype/config/webpack`.
+                                                                                                    
Your directory structure should look like this: 
```

File: packages/electrode-archetype-react-app/README.md [hunk 4]
```diff
@@ -236,6 +236,35 @@ var supports = require("electrode-archetype-react-app/supports");
 
 supports.cssModuleHook();
 ```
+## Extending Webpack Configuration
+
+The webpack config for your app is being generated by the archetype. While composing the webpack config, the archetype looks for the `archetype/config` folder at the project root of your app.
+In order to override or extend the webpack configuration you should create a file under `archetype/config/webpack`.
```
Comment (karlhorky) [inline packages/electrode-archetype-react-app/README.md]: Is this path `archetype/config/webpack` actually correct?

In https://github.com/electrode-io/electrode/issues/97#issuecomment-274150029 @ananavati mentioned that webpack configuration files in the root of the app will be used to extend the configuration. I've tested this to be working 2-3 weeks ago.

Has this been changed?

Comment (animesh10) [inline packages/electrode-archetype-react-app/README.md]: I think this change might have occurred around the same time, the way mentioned in the comment should still work, but this is the way for extending your webpack config, https://github.com/electrode-io/electrode/blob/master/packages/electrode-archetype-react-app-dev/config/webpack/util/generate-config.js#L15

Comment (karlhorky) [inline packages/electrode-archetype-react-app/README.md]: Oh nice, they both work! Do we want to mention both in the readme?

**Comment by seansean11eis** [comment]
@karlhorky thanks for adding this to the documentation.  Reading the docs, I'm still left wondering how one might override current webpack configuration.  Electrode is fairly opinionated and I think it is very likely that users will want to override settings in webpack (i.e. currently trying to turn off css modules).  I think that this section could be augmented.  Is there anywhere that you can find config override info?  Or should I be looking at the webpack-config-composer project.

**Comment by karlhorky** [comment]
@seansean11 Sorry that this is still unclear. In your project you would:

1. Add a folders structure like: `<PROJECT ROOT>/archetype/config/webpack`
2. Create a webpack configuration file matching the name of a file [here](https://github.com/electrode-io/electrode/tree/65884aa271ed76398c7bc062675d99627da94b8e/packages/electrode-archetype-react-app-dev/config/webpack) that you want to override (for instance, `webpack.config.dev.js` or `webpack.config.js`)
3. In this file, include the minimal configuration to override the key in the configuration that you want to affect. @ananavati mentions an example in https://github.com/electrode-io/electrode/issues/97#issuecomment-274150029. This will be merged by `webpack-config-composer`.

Does this explanation help?

**Comment by seansean11eis** [comment]
Hey @karlhorky thanks so much for the response.  I actually tried following the steps in @ananavati's comments before posting.  The electrode archetype has changed a bit since that comment.  I import the default settings like so:
``` js
const baseConfig = require("electrode-archetype-react-app-dev/config/webpack/webpack.config.js");
```
This require returns an empty object, so I'm a little confused on how to obtain default settings to override.

**Comment by jchip** [comment]
https://docs.electrode.io/chapter1/intermediate/app-archetype/webpack-config.html#overriding

**Pull request merged**